lib/explorer/backend/series.ex

```diff
@@ -238,6 +238,7 @@ defmodule Explorer.Backend.Series do
   @callback upcase(s) :: s
   @callback downcase(s) :: s
   @callback trim(s) :: s
+  @callback trim(s, String.t()) :: s
```

Comment:
I think we can keep only this callback, and remove the `trim(s)`. For what I could understand, you always pass an empty `""` by default, right? WDYT?

Reply:
Ahh ok, I wasn't sure if i would break it.

I pass an empty string but is that the best way? 

I don't really see how else i could encode it on the rust side unless it was an enum with an atom type and a string type.

like 

```rust
enum OptionalString {
     Atom(TermType::Atom),
     Str(String)
}
```
(not sure if this would work)

 and then pass in `:nil` when there is no string passed?

Reply:
Oh, we can have this as an optional value as well.
The callback could be like `trim(s, String.t() | nil)` and we could have a signature like this in the Rust side:

```rust
pub fn expr_trim(expr: ExExpr, string: Option<String>) -> ExExpr
```

Then you can call the native function with `nil` or a string, that Rustler is going to auto convert it to an `Option<String>` (you don't need a custom type for that). WDYT?

Reply:
That's exactly what i was hoping you would say 😆

Reply:
That's implemented now

Reply:
Cool! Now just run `mix ci` to fix the issues and we are good to go!

Reply:
getting a load of doctest failures, probably due to being on windows and the difference between \r\n and \n.

I'll get it running in a linux environment tomorrow and sort the remaining issues

Reply:
It's fine :)
If this helps: I ran the command here and the diff necessary to fix the issues is the following: 

```diff
diff --git a/lib/explorer/backend/lazy_series.ex b/lib/explorer/backend/lazy_series.ex
index 5a488d10..c33b6b52 100644
--- a/lib/explorer/backend/lazy_series.ex
+++ b/lib/explorer/backend/lazy_series.ex
@@ -113,7 +113,6 @@ defmodule Explorer.Backend.LazySeries do
     contains: 2,
     trim_leading: 1,
     trim_trailing: 1,
-    trim: 1,
     trim: 2,
     upcase: 1,
     downcase: 1,
diff --git a/lib/explorer/polars_backend/expression.ex b/lib/explorer/polars_backend/expression.ex
index 8e8aa781..d8c62c15 100644
--- a/lib/explorer/polars_backend/expression.ex
+++ b/lib/explorer/polars_backend/expression.ex
@@ -120,7 +120,6 @@ defmodule Explorer.PolarsBackend.Expression do
 
     # Strings
     contains: 2,
-    trim: 1,
     trim: 2,
     trim_leading: 1,
     trim_trailing: 1,
diff --git a/test/explorer/data_frame_test.exs b/test/explorer/data_frame_test.exs
index 7d6ba05a..60c8d1b5 100644
--- a/test/explorer/data_frame_test.exs
+++ b/test/explorer/data_frame_test.exs
@@ -1515,8 +1515,6 @@ defmodule Explorer.DataFrameTest do
     end
 
     test "trim characters from string" do
-      IO.puts("running trim characters")
-
       df =
         DF.new(
           a: ["£2", "3£", "£200£", "£££20"],
diff --git a/test/explorer/series_test.exs b/test/explorer/series_test.exs
index e140dadb..7225001d 100644
--- a/test/explorer/series_test.exs
+++ b/test/explorer/series_test.exs
@@ -3972,6 +3972,7 @@ defmodule Explorer.SeriesTest do
 
       assert Series.trim(series) |> Series.to_list() == ["123", "2", "20$"]
     end
+
     test "trim/2" do
       series = Series.from_list(["£1£23", "2£", "£20£"])
 
```